service/apigateway/internal/awsrestjson/marshal.go

```diff
@@ -0,0 +1,65 @@
+package awsrestjson
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/private/protocol/jsonrpc"
+	"github.com/aws/aws-sdk-go-v2/private/protocol/rest"
+	v2 "github.com/aws/aws-sdk-go-v2/private/protocol/rest/v2"
+	"github.com/aws/aws-sdk-go-v2/service/apigateway/types"
+)
+
+// ProtoGetAPIKeyMarshaler defines marshaler for ProtoGetApiKey operation
+type ProtoGetAPIKeyMarshaler struct {
+	Input *types.GetApiKeyInput
+}
+
+// MarshalOperation is the top level method used within a handler stack to marshal an operation
+// This method calls appropriate marshal shape functions as per the input shape and protocol used by the service.
+func (m ProtoGetAPIKeyMarshaler) MarshalOperation(r *aws.Request) {
+	var err error
+	encoder := v2.NewEncoder(r.HTTPRequest)
+	// adds content-type header
+	encoder.AddHeader("Content-Type").String("application/json")
+
+	err = MarshalGetAPIKeyInputShapeAWSREST(m.Input, encoder)
+	if err != nil {
+		r.Error = err
+	}
+
+	err = MarshalGetAPIKeyInputShapeAWSJSON(m.Input, r)
```

Comment:
It would be helpful here to add a TODO that this needs to be updated not to pass in `r`  directly but a payload